internal/services/nss/nss.go

```diff
@@ -161,6 +161,40 @@ func (s Service) GetShadowEntries(ctx context.Context, req *authd.Empty) (*authd
 	return &r, nil
 }
 
+// DisableUser marks a user as disabled.
+func (s Service) DisableUser(ctx context.Context, req *authd.DisableUserRequest) (*authd.Empty, error) {
+	if err := s.permissionManager.IsRequestFromRoot(ctx); err != nil {
+		return nil, err
+	}
+
+	if req.GetName() == "" {
+		return nil, status.Error(codes.InvalidArgument, "no user name provided")
+	}
+
+	if err := s.userManager.DisableUser(req.GetName()); err != nil {
+		return nil, err
+	}
+
+	return &authd.Empty{}, nil
+}
+
+// EnableUser marks a user as enabled.
+func (s Service) EnableUser(ctx context.Context, req *authd.EnableUserRequest) (*authd.Empty, error) {
+	if err := s.permissionManager.IsRequestFromRoot(ctx); err != nil {
+		return nil, err
+	}
+
+	if req.GetName() == "" {
+		return nil, status.Error(codes.InvalidArgument, "no user name provided")
+	}
+
+	if err := s.userManager.EnableUser(req.GetName()); err != nil {
+		return nil, err
+	}
+
+	return &authd.Empty{}, nil
+}
+
```

Comment:
I'm a bit unsure this should be part of the NSS service...

The nss service is meant to be the one for NSS operations and these do not include enabling/disabling an user AFAIK, so this should be IMHO part of an authd-management specific service, rather than in the NSS one, that is the one to be used by the NSS module to provide the group and passwd information.

@didrocks any opinion here?

Reply:
Ah I saw you had the same concerns: https://github.com/ubuntu/authd/issues/640#issuecomment-2616291629

So we should move this out from NSS and have another management service, let's define a good name then.

Reply:
Yes, we should decide how we want to restructure the service. I made some suggestions on https://github.com/ubuntu/authd/issues/640#issuecomment-2616326495. I'm not sure we have to block merging this PR on that. I don't expect @shiv-tyagi to do that work, IMO we could do it as a follow-up.

Reply:
Isn't better to make it before and then rebase this PR on that?

Reply:
We can also do that.

Reply:
If we are doing a PR for restructuring this, I will wait till it is merged and then rebase my work on top of it.

Reply:
> So we should move this out from NSS and have another management service, let's define a good name then.

@adombeck @3v1n0 Any update on this?

Reply:
Yes, see https://github.com/ubuntu/authd/pull/855

Reply:
Oh great! Thanks @adombeck. Looking forward.